geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeRules.java

```diff
@@ -0,0 +1,375 @@
+      return inFields.get(inputRef.getIndex());
+    }
+
+    @Override public String visitCall(RexCall call) {
+      final List<String> strings = visitList(call.operands);
+      if (call.getOperator() == SqlStdOperatorTable.ITEM) {
+        final RexNode op1 = call.getOperands().get(1);
+        if (op1 instanceof RexLiteral) {
+          if (op1.getType().getSqlTypeName() == SqlTypeName.INTEGER) {
+            return stripQuotes(strings.get(0)) + "[" + ((RexLiteral) op1).getValue2() + "]";
+          } else if (op1.getType().getSqlTypeName() == SqlTypeName.CHAR) {
+            return stripQuotes(strings.get(0)) + "." + ((RexLiteral) op1).getValue2();
+          }
+        }
+      }
+
+      return super.visitCall(call);
+    }
+
+    private String stripQuotes(String s) {
+      return s.startsWith("'") && s.endsWith("'") ? s.substring(1, s.length() - 1) : s;
+    }
+
+    List<String> visitList(List<RexNode> list) {
+      final List<String> strings = new ArrayList<>();
+      for (RexNode node : list) {
+        strings.add(node.accept(this));
+      }
+      return strings;
+    }
+  }
+
+  /**
+   * Rule to convert a {@link LogicalProject} to a {@link GeodeProjectRel}.
+   */
+  private static class GeodeProjectRule extends GeodeConverterRule {
+
+    private static final GeodeProjectRule INSTANCE = new GeodeProjectRule();
+
+    private GeodeProjectRule() {
+      super(LogicalProject.class, "GeodeProjectRule");
+    }
+
+    @Override public boolean matches(RelOptRuleCall call) {
+      LogicalProject project = call.rel(0);
+      for (RexNode e : project.getProjects()) {
+        if (e.getType().getSqlTypeName() == SqlTypeName.GEOMETRY) {
+          // For spatial Functions Drop to Calcite Enumerable
+          return false;
+        }
+      }
+
+      return true;
+    }
+
+    @Override public RelNode convert(RelNode rel) {
+      final LogicalProject project = (LogicalProject) rel;
+      final RelTraitSet traitSet = project.getTraitSet().replace(out);
+      return new GeodeProjectRel(
+          project.getCluster(),
+          traitSet,
+          convert(project.getInput(), out),
+          project.getProjects(),
+          project.getRowType());
+    }
+  }
+
+  /**
+   * Rule to convert the Limit in {@link org.apache.calcite.rel.core.Aggregate} to a
+   * {@link GeodeAggregateRel}.
+   */
+  private static class GeodeAggregateRule extends GeodeConverterRule {
+
+    private static final GeodeAggregateRule INSTANCE = new GeodeAggregateRule();
+
+    GeodeAggregateRule() {
+      super(LogicalAggregate.class, "GeodeAggregateRule");
+    }
+
+    @Override public RelNode convert(RelNode rel) {
+      final LogicalAggregate aggregate = (LogicalAggregate) rel;
+      final RelTraitSet traitSet = aggregate.getTraitSet().replace(out);
+      return new GeodeAggregateRel(
+          aggregate.getCluster(),
+          traitSet,
+          convert(aggregate.getInput(), traitSet.simplify()), // or , out?
```

Comment:
Is this something which still needs to be addressed or perhaps improved on in the future? It would be helpful to have a more detailed comment if that's the case.

Reply:
+1 just removed it